tools/packaging/packaging.psm1

```diff
@@ -745,7 +733,70 @@ function New-UnixPackage {
     }
 }
 
+function New-MacOsDistributionPackage
+{
+    param(
+        [Parameter(Mandatory,HelpMessage='The FileInfo of the file created by FPM')]
+        [System.IO.FileInfo]$FpmPackage
+    )
+
+    if(!$Environment.IsMacOS)
+    {
+        throw 'New-MacOsDistributionPackage is only supported on macOS!'
+    }
+
+    $packageName = Split-Path -leaf -Path $FpmPackage
+
+    # Create a temp directory to store the needed files
+    $tempDir = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+    New-Item -ItemType Directory -Path $tempDir -Force > $null
+
+    $resourcesDir = Join-Path -path $tempDir -childPath 'resources'
+    New-Item -ItemType Directory -Path $resourcesDir -Force > $null
+    #Copy background file to temp directory
+    $backgroundFile = Join-Path $PSScriptRoot "/../../assets/macDialog.png"
+    Copy-Item -Path $backgroundFile -Destination $resourcesDir
+    # Move the current package to the temp directory
+    $tempPackagePath = Join-Path -path $tempDir -ChildPath $packageName
+    Move-Item -Path $FpmPackage -Destination $tempPackagePath -Force
+
+    # Add the OS information to the macOS package file name.
+    $packageExt = [System.IO.Path]::GetExtension($FpmPackage.Name)
+    $packageNameWithoutExt = [System.IO.Path]::GetFileNameWithoutExtension($FpmPackage.Name)
+
+    $newPackageName = "{0}-{1}{2}" -f $packageNameWithoutExt, $script:Options.Runtime, $packageExt
+    $newPackagePath = Join-Path $FpmPackage.DirectoryName $newPackageName
+
+    # -Force is not deleting the NewName if it exists, so delete it if it does
+    if ($Force -and (Test-Path -Path $newPackagePath))
+    {
+        Remove-Item -Force $newPackagePath
+    }
+
+    # Create the distribution xml
+    $distributionXmlPath = Join-Path -Path $tempDir -ChildPath 'powershellDistribution.xml'
 
+    # format distribution template with:
+    # 0 - title
+    # 1 - version
+    # 2 - package path
+    # 2 - minimum os version
+    $PackagingStrings.OsxDistributionTemplate -f "PowerShell - $Version", $Version, $packageName, '10.12' | Out-File -Encoding ascii -FilePath $distributionXmlPath -Force
+
+    log "Applying distribution.xml to package..."
+    Push-Location $tempDir
+    try
+    {
+        Start-NativeExecution -sb {productbuild --distribution $distributionXmlPath --resources $resourcesDir $newPackagePath}
```

Comment:
Non-blocking: it's better to add a comment explaining that `productbuild is a xcode command line tool which will be installed when homebrew is installed`.